File: Activity001/my_cpp_pkg/src/my_first_node.cpp
#include "rclcpp/rclcpp.hpp"

class MyNode: public rclcpp::Node
{
private:
    void timerCallBack(void){
        counter++;
        RCLCPP_INFO(this->get_logger(), "Hello %d", counter);
    }
    rclcpp::TimerBase::SharedPtr timer;
public:
    int counter;
    MyNode():Node("CPP_Test"){
        counter = 0;
        RCLCPP_INFO(this->get_logger(), "Hello ROS2 with CPP");
    }

    void printHello(void){
        timer = this->create_wall_timer(std::chrono::milliseconds(500) , std::bind(&MyNode::timerCallBack, this));
    }
};



int main(int argc, char** argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<MyNode>();
    node->printHello();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: Activity001/my_py_pkg/my_py_pkg/my_first_node.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node

class MyNode(Node):
    def __init__(self):
        super().__init__("py_test")
        self.counter = 0
        self.get_logger().info("Hello ROS2 with OOP")
        self.create_timer(1, self.timer_callBack)
    
    def timer_callBack(self):
        self.counter += 1
        self.get_logger().info("Hello " + str(self.counter))
        

def main(args=None):
    rclpy.init(args=args)
    node = MyNode()
    #node.get_logger().info("Hello ROS2")
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()
    